launcher/modManager/cmodlistview_moc.cpp

```diff
@@ -870,7 +879,35 @@ void CModListView::installFiles(QStringList files)
 		QString realFilename = Helper::getRealPath(filename);
 
 		if(realFilename.endsWith(".zip", Qt::CaseInsensitive))
-			mods.push_back(filename);
+		{
+			ZipArchive archive(qstringToPath(realFilename));
+			auto fileList = archive.listFiles();
+
+			bool hasModJson = false;
+			bool hasMaps = false;
+
+			for (const auto& file : fileList)
+			{
+				QString lower = QString::fromStdString(file).toLower();
+
+				// Check for mod.json anywhere in archive
+				if (lower.endsWith("mod.json"))
+				{
+					hasModJson = true;
+					break;
+				}
+
+				// Check for map files anywhere
+				if (lower.endsWith(".h3m") || lower.endsWith(".h3c") || lower.endsWith(".vmap") || lower.endsWith(".vcmp"))
+					hasMaps = true;
```

Comment:
What if we're importing *mod* with maps, e.g. Antagarich Burning or any map pack?

In this case it looks like we'll install 2 copies of maps - into Maps directory and as part of the mod

Reply:
Are you sure? When `hasModJson` is set, it would break out of the loop. But what if map was found before mod.json? In that case, both `hasModJson` and `hasMaps` will be set

Reply:
Fixed conditions, should be fine now.